dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/importers/ui/ImportersForView.java

```diff
@@ -0,0 +1,82 @@
+package edu.harvard.iq.dataverse.importers.ui;
+
+import edu.harvard.iq.dataverse.importer.metadata.ImporterConstants;
+import edu.harvard.iq.dataverse.importer.metadata.MetadataImporter;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.MetadataBlock;
+import edu.harvard.iq.dataverse.persistence.dataverse.Dataverse;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ImportersForView implements MapForView<MetadataImporter, ImportersForView.ImporterItem> {
+    private Map<MetadataImporter, ImporterItem> items = new HashMap<>();
+
+    // -------------------- GETTERS --------------------
+
+    public List<MetadataImporter> getImportersView() {
+        return new ArrayList<>(items.keySet()); // TODO: Implement some kind of sorting later
+    }
+
+    @Override
+    public Map<MetadataImporter, ImporterItem> getUnderlyingMap() {
+        return items;
+    }
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    public ImportersForView(Dataset dataset, Map<String, MetadataImporter> importers, Locale locale) {
+
+        if (dataset == null || locale == null) {
+            return;
+        }
+
+        Dataverse owner = dataset.getOwner();
+        Set<String> metadataBlockNames = owner.getMetadataBlocks().stream()
```

Comment:
Dataverse can take metadata blocks from it's parent. Then method will return empty list.
Consider using `getRootMetadataBlocks` or `getMetadataBlockRootDataverse`

Reply:
Ok, but does it mean that we should check:
1. Only root
2. Root + child
3. Root + child + every dataverse between?

Reply:
We should obtain 'active' metadata blocks from first ancestor (parent, parent of parent, ... etc.) that defines it's own metadata blocks (dataverse.metadataBlockRoot is set to true). And only that dataverse.